oracle_cardano/processor/txs_processor/data.go

```diff
@@ -58,9 +58,17 @@ func (pc *txProcessorsCollection) getSuccess(tx *core.CardanoTx, appConfig *cCor
 			return nil, err
 		}
 
+		if !appConfig.RefundEnabled && metadata.BridgingTxType == common.TxTypeRefundRequest {
+			return nil, fmt.Errorf("irrelevant tx. Tx type: %s", metadata.BridgingTxType)
+		}
+
 		txProcessor, relevant = pc.successTxProcessors[string(metadata.BridgingTxType)]
 		if !relevant {
-			txProcessor = pc.successTxProcessors[string(common.TxTypeRefundRequest)]
+			if appConfig.RefundEnabled {
+				txProcessor = pc.successTxProcessors[string(common.TxTypeRefundRequest)]
```

Comment:
maybe we should have  `RefundNotEnabledRequestProcessor `instantiated in the case of `!RefundEnabled`. than we do not have if here and also here:
```
if !appConfig.RefundEnabled && metadata.BridgingTxType == common.TxTypeRefundRequest {
			return nil, fmt.Errorf("irrelevant tx. Tx type: %s", metadata.BridgingTxType)
}
``` (error can be returned from that new processor)
Also this:
```
if !appConfig.RefundEnabled {
		return fmt.Errorf("refund is not enabled")
	}
```

We can even put check 
```
tx.BatchTryCount > appConfig.TryCountLimits.MaxBatchTryCount ||
		tx.SubmitTryCount > appConfig.TryCountLimits.MaxSubmitTryCount) 
```

inside those two processors and remove if enabled in callers